lib/Backend/IRBuilder.cpp

```diff
@@ -6074,7 +6074,24 @@ IRBuilder::BuildProfiledCallI(Js::OpCode opcode, uint32 offset, Js::RegSlot retu
         newOpcode = opcode;
         Js::OpCodeUtil::ConvertNonCallOpToNonProfiled(newOpcode);
         Assert(newOpcode == Js::OpCode::NewScObject || newOpcode == Js::OpCode::NewScObjectSpread);
-        returnType = ValueType::GetObject(ObjectType::UninitializedObject);
+        if (newOpcode == Js::OpCode::NewScObjectSpread || !this->m_func->HasProfileInfo())
+        {
+            // Testing required before enabling this for newscobjectspread
+            returnType = ValueType::GetObject(ObjectType::UninitializedObject);
+        }
+        else
+        {
+            // If we have profile data, make use of it
+            ValueType knownReturnType = this->m_func->GetReadOnlyProfileInfo()->GetReturnType(opcode, profileId);
+            if (knownReturnType != ValueType::GetObject(ObjectType::UninitializedObject))
+            {
+                returnType = knownReturnType;
+            }
+            else
+            {
+                returnType = ValueType::GetObject(ObjectType::UninitializedObject);
+            }
```

Comment:
Can we just have "returnType = knownReturnType" or "returnType = this->m_func->GetReadOnlyProfileInfo()->GetReturnType(opcode, profileId)" instead of the if-else?